src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Reader/CborReader.cs

```diff
@@ -81,6 +81,34 @@ public ReadOnlyMemory<byte> ReadEncodedValue(bool disableConformanceModeChecks =
             return _data.Slice(initialOffset, _offset - initialOffset);
         }
 
+        /// <summary>
+        /// Resets the <see cref="CborReader"/> instance over the specified <paramref name="data"/> with unchanged configuration.
+        /// <see cref="ConformanceMode"/> and <see cref="AllowMultipleRootLevelValues"/> are unchanged.
+        /// </summary>
+        /// <param name="data">The CBOR-encoded data to read.</param>
+        public void Reset(ReadOnlyMemory<byte> data)
+        {
+            // ConformanceMode and AllowMultipleRootLevelValues are set in ctor, they remain unchanged.
+
+            _data = data;
+            _offset = 0;
+
+            _nestedDataItems?.Clear();
+            _currentMajorType = default;
+            _definiteLength = AllowMultipleRootLevelValues ? null : 1;
+            _itemsRead = default;
+            _frameOffset = default;
+            _isTagContext = default;
+            _currentKeyOffset = default;
+            _previousKeyEncodingRange = default;
+            _keyEncodingRanges?.Clear();
+            _isConformanceModeCheckEnabled = true;
+            _cachedState = CborReaderState.Undefined;
+
+            //_pooledKeyEncodingRangeAllocations?.Clear(); // We don't need to clear these previously created and reusable instances
+            //_indefiniteLengthStringRangeAllocation?.Clear(); // We don't need to clear this previously created and reused instance
```

Comment:
Please don't check in commented out code.  The trailing comments themselves are useful, but need to be adjusted to the code not being there.  So something like

```suggestion
            // We don't need to clear the reusable instances in _pooledKeyEncodingRangeAllocations
            // or _indefiniteLengthStringRangeAllocation.
```